internal/k8s/status.go

```diff
@@ -0,0 +1,65 @@
+// Copyright © 2018 Heptio
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"encoding/json"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	ingressroutev1 "github.com/heptio/contour/apis/contour/v1beta1"
+	clientset "github.com/heptio/contour/internal/generated/clientset/versioned"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// IngressRouteStatus allows for updating the object's Status field
+type IngressRouteStatus struct {
+	Client clientset.Interface
+}
+
+// SetStatus sets the IngressRoute status field to an Valid or Invalid status
+func (irs *IngressRouteStatus) SetStatus(msg, desc string, existing *ingressroutev1.IngressRoute) error {
```

Comment:
nit: thoughts on renaming `msg` to `status`?

Reply:
I renamed `msg` to `status` and did that back through the dag as well.